First bitsurance workflow integration

Beerosagos opened the pull request:
This creates the Insurance menu item, that leads to the bitsurance workflow
starting page. From there, it is possible to detect already insured accounts
or begin the onramp of a new account insurance. A bitsurance Iframe is
integrated in bitsurance/widget.tsx, with a structure which is very similar
to the Pocket one.

Please note:
- the widget page is just a draft and will probably be heavily modified once the iframe widget is ready.
- atm the T&C page is just the Pocket one, I'll create a new one when the text will be ready.
- CSS needs to be improved ofc, It will be the last step to avoid rewriting code :) 

Next step after merging this: redirect insured accounts to a dedicated dashboard.

Sorry for not splitting in more commits, hope it won't be too hard to review :pray:

Beerosagos commented on backend/accounts.go:
```diff
@@ -213,7 +213,7 @@ func (backend *Backend) LookupInsuredAccounts(accountCode accountsTypes.Code) ([
 		for _, account := range backend.accounts {
 			config := account.Config().Config
-			if !config.Inactive && !config.HiddenBecauseUnused && (config.CoinCode == coinpkg.CodeTBTC || config.CoinCode == coinpkg.CodeBTC) {
+			if !config.HiddenBecauseUnused && (config.CoinCode == coinpkg.CodeTBTC || config.CoinCode == coinpkg.CodeBTC) {
```
We need to be aware of insured accounts also when hidden

Beerosagos commented on backend/bitsurance/bitsurance.go:
```diff
@@ -54,7 +54,7 @@ func bitsuranceCheckId(httpClient *http.Client, accountId string) (bool, error)
 	details := accountDetails{}
 	code, err := util.APIGet(httpClient, endpoint, apiKey, 1024, &details)
-	if err != nil {
+	if err != nil && code != http.StatusNotFound {
```
404 is not a real error in this case, as it is the expected response in case an account is not insured.

Beerosagos commented on backend/bitsurance/bitsurance.go:
```diff
@@ -83,11 +83,11 @@ func BitsuranceGetId(account accounts.Interface) (string, error) {
 // BitsuranceURL returns the url for the Bitsurance widget for a given locale.
-func BitsuranceURL(lang, bitsuranceId string) string {
-	return widgetURL + lang + "/?id=" + bitsuranceId
+func BitsuranceURL(lang string) string {
```
We don't need the bitsuranceId when calling the iframe, as we decided to use it only for the onramp of new accounts

thisconnect reviewed (commented):
frontend looks good, I added some comments, but I think most is things we can refactor later.

btw. is there a testnet insurance or any other way to try it?

Beerosagos commented on frontends/web/src/routes/bitsurance/account.tsx:
```diff
@@ -0,0 +1,101 @@
+import { useTranslation } from 'react-i18next';
+import { route } from '../../utils/route';
+import { IAccount } from '../../api/account';
+import Guide from './guide';
```
Done, thanks

Beerosagos commented on frontends/web/src/routes/bitsurance/bitsurance.tsx:
```diff
@@ -41,25 +51,79 @@ export const Bitsurance = ({ accounts }: TProps) => {
+      if (response.accountCodes.includes(account.code)) {
+        setInsuredAccounts(insuredAccounts => [...insuredAccounts, account]);
+      }
+    }
```
nice! thanks. Done to same for the previous 'useEffect' too

Beerosagos commented on frontends/web/src/routes/bitsurance/widget.tsx:
```diff
@@ -0,0 +1,254 @@
+
+interface TProps {
+    code: string;
+}
```
Whoops, copy&paste gone wrong :D

Beerosagos commented on frontends/web/src/routes/bitsurance/widget.tsx:
```diff
@@ -0,0 +1,254 @@
+      case V0MessageType.RequestAddress:
+        if (!signing) {
+          handleRequestAddress(message);
+        }
```
Added a comment, it's basically used to ignore new signing requests while there is one still ongoing

Beerosagos commented on frontends/web/src/routes/bitsurance/widget.tsx:
```diff
@@ -0,0 +1,254 @@
+      }
+      setHeight(ref.current.offsetHeight);
+    }, 200);
+  };
```
Ok, didn't touch it for now

Beerosagos commented:
> frontend looks good, I added some comments, but I think most is things we can refactor later.
> 
> btw. is there a testnet insurance or any other way to try it?

Thanks, I fixed most of it. Insurance only works on mainnet, but if you compile with 'make servewallet' it goes on the staging server so you can test without worries (the iframe development is still ongoing, tho). To go on production server you'll need 'make servewallet-prodservers`.

Ping @benma for the backend :pray:

benma commented on backend/handlers/handlers.go:
```diff
@@ -1100,7 +1100,7 @@ func (handlers *Handlers) getBitsuranceURL(r *http.Request) interface{} {
 	}
 
-	return bitsurance.BitsuranceURL(lang, r.URL.Query().Get("code"))
+	return bitsurance.BitsuranceURL(handlers.backend.DevServers(), lang)
```
makes me think of testnet - i you run 'BitBox -testnet`, what is the expected behavior for bitsurance? 

in `BitsuranceURL()` TBTC seems enabled, but that does not make much sense for users right?

Beerosagos replied:
Good point, I removed the check for testnet accounts as they won't be supported. In testnet mode the insurance page will be available, but there will be no insurable accounts on the account selection page. I updated the last commit message to specify that testnet is not supported.

benma reviewed (approved):
backend utACK, but with a question around testnet mode (which endusers can access)
